packages/netlify-cms-widget-markdown/src/MarkdownControl/plugins/Link.js

```diff
@@ -2,24 +2,31 @@ function Link({ type }) {
   return {
     commands: {
       toggleLink(editor, getUrl) {
+        const selection = editor.value.selection;
+        const isCollapsed = selection && selection.isCollapsed;
+
         if (editor.hasInline(type)) {
-          editor.unwrapInline(type);
+          const inlines = editor.value.inlines.toJSON();
+          const link = inlines.find(item => item.type === type);
+
+          const url = getUrl(link.data.url);
+
+          // remove url if it was removed by the user
+          if (url === '') editor.unwrapInline(type);
+
+          // if selection is empty, replace the old link
```

Comment:
Any reason to do this only when the selection is empty?

Reply:
Ok, so at the moment the link gets deleted before the prompt opens, with this PR the link will get deleted _only_ when the user explicitly deletes it in the prompt.

I didn't see any other case when you would remove the link. Do you think there should be other checks for links? Like if it's a path or a full URL?

Reply:
I think I'm misunderstanding something won't `if (url && isCollapsed) editor.setInlines({ data: { url } })`  only allow users to edit unselected links?

Reply:
Hmmm, I think yes, removing isCollapsed should allow editing only partly selected links. I'll take a look again in a bit.

Reply:
Yep, you were right. I also added a small css change. Now links will inherit the font-size of the parent element (e.g. in a heading).